### ISSUE TITLE
Wrong assumption about Docker environment

### ISSUE THREAD

**Issue by sameo** [01org/cc-oci-runtime#401]
When running COR through [CRI-O](https://github.com/kubernetes-incubator/cri-o), we get the following config.json:

```
{
	"ociVersion": "1.0.0-rc1",
	"platform": {
		"os": "linux",
		"arch": "amd64"
	},
	"process": {
		"user": {
			"uid": 0,
			"gid": 0
		},
		"args": [
			"/pause"
		],
		"env": [
			"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
			"TERM=xterm"
		],
		"cwd": "/",
		"capabilities": [
			"CAP_CHOWN",
			"CAP_DAC_OVERRIDE",
			"CAP_FSETID",
			"CAP_FOWNER",
			"CAP_MKNOD",
			"CAP_NET_RAW",
			"CAP_SETGID",
			"CAP_SETUID",
			"CAP_SETFCAP",
			"CAP_SETPCAP",
			"CAP_NET_BIND_SERVICE",
			"CAP_SYS_CHROOT",
			"CAP_KILL",
			"CAP_AUDIT_WRITE"
		],
		"rlimits": [
			{
				"type": "RLIMIT_NOFILE",
				"hard": 1024,
				"soft": 1024
			}
		]
	},
	"root": {
		"path": "/var/lib/ocid/graph/vfs/pause/rootfs",
		"readonly": true
	},
	"hostname": "ocic_host",
	"mounts": [
		{
			"destination": "/proc",
			"type": "proc",
			"source": "proc"
		},
		{
			"destination": "/dev",
			"type": "tmpfs",
			"source": "tmpfs",
			"options": [
				"nosuid",
				"strictatime",
				"mode=755",
				"size=65536k"
			]
		},
		{
			"destination": "/dev/pts",
			"type": "devpts",
			"source": "devpts",
			"options": [
				"nosuid",
				"noexec",
				"newinstance",
				"ptmxmode=0666",
				"mode=0620",
				"gid=5"
			]
		},
		{
			"destination": "/dev/shm",
			"type": "tmpfs",
			"source": "shm",
			"options": [
				"nosuid",
				"noexec",
				"nodev",
				"mode=1777",
				"size=65536k"
			]
		},
		{
			"destination": "/dev/mqueue",
			"type": "mqueue",
			"source": "mqueue",
			"options": [
				"nosuid",
				"noexec",
				"nodev"
			]
		},
		{
			"destination": "/sys",
			"type": "sysfs",
			"source": "sysfs",
			"options": [
				"nosuid",
				"noexec",
				"nodev",
				"ro"
			]
		},
		{
			"destination": "/etc/resolv.conf",
			"type": "bind",
			"source": "/var/lib/ocid/sandboxes/643c5fc887c5121563fb7d001c610a34f440adf371418467046350336d3a5a3d/resolv.conf",
			"options": [
				"bind",
				"ro"
			]
		}
	],
	"hooks": {},
	"annotations": {
		"ocid/annotations": "{\"owner\":\"hmeng\"}",
		"ocid/container_id": "f22a77addbd5e7465b3a25308641297ba4ec7c3ad8ecab79a5f5e8748ce2d319",
		"ocid/container_name": "otc-6.test.ocid-podsandbox1-1-infra",
		"ocid/labels": "{\"group\":\"test\"}",
		"ocid/log_path": ".",
		"ocid/metadata": "{\"name\":\"podsandbox1\",\"uid\":\"redhat-test-ocid\",\"namespace\":\"otc-6.test.ocid\",\"attempt\":1}",
		"ocid/name": "otc-6.test.ocid-podsandbox1-1",
		"owner": "hmeng"
	},
	"linux": {
		"resources": {
			"devices": [
				{
					"allow": false,
					"access": "rwm"
				}
			]
		},
		"cgroupsPath": "podsandbox1.slice:container:infra",
		"namespaces": [
			{
				"type": "pid"
			},
			{
				"type": "network"
			},
			{
				"type": "ipc"
			},
			{
				"type": "uts"
			},
			{
				"type": "mount"
			}
		]
	}
}
```

where process.terminal is not set. That brings COR into creating a qemu chardev pipe linking to the dockerd named pipes. With CRI-O, there is no dockerd and we fail with e.g. the following error:

```
 qemu-lite-system-x86_64: -chardev pipe,id=charconsole0,path=/var/lib/ocid/sandboxes/3f2216b5532dc804ae13fe8e9675221f8584fc8ac2dc3c232efe2dcfa0515608/cc-std: Could not open '/var/lib/ocid/sandboxes/3f2216b5532dc804ae13fe8e9675221f8584fc8ac2dc3c232efe2dcfa0515608/cc-std': No such file or directory
```

**Comment by amshinde** [comment]
@sameo This should be fixed with hyperstart integration work as we are moving away from creating the links to the docker named pipes.

**Comment by sameo** [comment]
@amshinde Do you mind this piece of code will be removed or replaced by something else ?

**Comment by amshinde** [comment]
Yes. This piece of code will be removed from the runtime. The shim will be handling the stdio without any of the docker pipe hacks. The shim simply writes to its stdout and this will be routed correctly to docker.

**Comment by amshinde** [comment]
This has been fixed in the 2.1 branch.
